src/chapters/classes-part2/customize-fields.rst

```diff
@@ -0,0 +1,283 @@
+      System.out.println("Absolute zero in F is: " + Temperature.absoluteZeroFahrenheit);
+
+      // Or if we have an object named "temp" of type Temperature
+      System.out.println("Absolute zero in F is: " + temp.absoluteZeroFahrenheit);
+
+When accessing a field from outside of its class, line 2 shows the preferred
+technique. The syntax makes it explicit that the field  is ``static``. Line 5
+does not make this point clear.
+
+.. admonition:: Example
+
+   As another example, we might also provide a third constructor for our
+   ``Student`` class that only requires the student’s name. Theoretically, the
+   ``studentId`` would (or could) be generated by the class itself.
+
+   .. sourcecode:: java
+      :linenos:
+
+      public class Student {
+
+         private static int nextStudentId = 1;
+         private String name;
+         private final int studentId;
+         private int numberOfCredits;
+         private double gpa;
+
+         public Student(String name, int studentId,
+                  int numberOfCredits, double gpa) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = numberOfCredits;
+            this.gpa = gpa;
+         }
+
+         public Student(String name, int studentId) {
+            this(name, studentId, 0, 0);
+         }
+
+         public Student(String name) {
+            this(name, nextStudentId);
+            nextStudentId++;
+         }
+
+         /* getters and setters omitted */
+
+      }
+
+In line 3, we add a static integer field that will keep track of the next
+student ID to be assigned to a student. Then, our new constructor (line 21)
+takes only a name as a parameter and assigns the student the next available ID.
+This works because static fields are shared across all objects created from
+the ``Student`` class, so it functions as a counter of sorts for the number of
+``Student`` objects created.
+
+Constants
+----------
+
+Unlike some other languages, Java doesn’t have a special keyword to declare a
+constant, or unchanging, variable. However, we can achieve the same result
+using a combination of ``static`` and ``final``.
+
+.. sourcecode:: java
+   :linenos:
+
+   public class Constants {
+      public static final double PI = 3.14159;
+      public static final String FIRST_PRESIDENT = "George Washington";
+   }
+
+Throughout the rest of this course, when we say *constant* we will mean a
+``static final`` variable.
+
+Three things to note from this example:
+
+#. We use a different naming convention for constants than for other variables.
+   Constants should be in ALL CAPS, with an underscore to separate words.
+#. There is no strong reason to make constants ``private``, since restricting
+   access would force us to re-declare the same values in different classes.
+   We’ll generally make our constants ``public``.
+#. Declare and initialize constants at the same time. Otherwise, you run the
```

Comment:
Hmmm... Let's see if I can transcribe my thinking.

1) Here is the original text from the old curriculum, "There is no strong reason to make constants private, as long as we initialize them where they are declared (so that somebody else doesn't give them a value first!). We'll generally make our constants public."

2) Then I remembered this final field example:
```
public final double doubleValue;

// This is allowed since we haven't initialized doubleValue yet
demo.doubleValue = 42.0;
```

3) Based on the example and the old text, it seems like you can declare a constant value without initializing it. However, you SHOULD, because otherwise the constant could be set in a later step, and that value might not be the one you want.
4) I updated the old text to try to make it flow better.

Please offer your own insights.

Reply:
I think the old text is wrong on this. I looked through available resources on the internet and the consensus from Java programmers is that a constant or ``static final`` variable must be declared and initialized in one line. There might be a few outlier cases where you can do this, but those circumstances are not something that is within the scope of this class.

Reply:
Got it.  Fixed as per your original comment.